src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -315,6 +393,27 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
           navigationOpen={navigationOpen}
           navigationWidth={navigationWidth}
           tools={activeDrawer && <AppLayoutDrawer appLayoutInternals={appLayoutInternals} />}
+          globalTools={
+            !!globalDrawers.length && (
+              <ActiveDrawersContext.Provider value={activeGlobalDrawersIds}>
+                {globalDrawers
+                  .filter(
+                    drawer =>
+                      activeGlobalDrawersIds.includes(drawer.id) ||
+                      (drawer.preserveInactiveContent && openDrawersHistory.has(drawer.id))
```

Comment:
**Before merge:**

Instead of pulling `openDrawersHistory` from useDrawers, how about tracking this state inside `AppLayoutGlobalDrawers` component?

Reply:
done